### PR TITLE
CIRC-1497 Get fixed period for loan due date during recall

### PR DESCRIPTION
**Pull request by Mykyta-Varenyk**
Resolves: https://issues.folio.org/browse/CIRC-1497

## Purpose
When recalling loans with fixed due date loan policies that have been renewed in a new loan period or had their due dates changed to a date after the original due date, the loan's due date is being set to the due date specified for the original fixed due date period for the loan's loanDate.
## Approach
When fetching fixed period for deciding on the due date, get a schedule in which a loan due date falls into.

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/policy/library/ClosedLibraryStrategyService.java [hunk 1]
```diff
@@ -107,10 +118,10 @@ public ClosedLibraryStrategyService(
 
   private Result<ZonedDateTime> applyFixedDueDateLimit(
     ZonedDateTime dueDate, Loan loan, LoanPolicy loanPolicy, AdjacentOpeningDays openingDays,
-    ZoneId timeZone) {
+    ZoneId timeZone, boolean isRecall){
 
     Optional<ZonedDateTime> optionalDueDateLimit =
-      loanPolicy.getScheduleLimit(loan.getLoanDate(), isRenewal, currentDateTime);
+      loanPolicy.getScheduleLimit(isRecall ? loan.getDueDate() : loan.getLoanDate(), isRenewal, currentDateTime);
```
Comment (sherzodn) [inline src/main/java/org/folio/circulation/domain/policy/library/ClosedLibraryStrategyService.java:124]: more than 100 chars in this row

File: src/test/java/api/requests/scenarios/RecallItemsTests.java [hunk 2]
```diff
@@ -8,14 +8,25 @@
 import static org.folio.circulation.support.utils.DateFormatUtil.formatDateTime;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.is;
+
+import java.time.Clock;
+import java.time.Instant;
```
Comment (sherzodn) [inline src/test/java/api/requests/scenarios/RecallItemsTests.java:14]: unused import

File: src/test/java/api/requests/scenarios/RecallItemsTests.java [hunk 3]
```diff
@@ -8,14 +8,25 @@
...
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.temporal.TemporalAccessor;
```
Comment (sherzodn) [inline src/test/java/api/requests/scenarios/RecallItemsTests.java:17]: the same here

**Review by sherzodn — APPROVED** [review]

File: src/main/java/org/folio/circulation/domain/policy/library/ClosedLibraryStrategyService.java (hunk 1 above)
Comment (Mykyta-Varenyk) [reply src/main/java/org/folio/circulation/domain/policy/library/ClosedLibraryStrategyService.java]: Fixed

File: src/test/java/api/requests/scenarios/RecallItemsTests.java (hunk 2 above)
Comment (Mykyta-Varenyk) [reply src/test/java/api/requests/scenarios/RecallItemsTests.java]: Fixed

File: src/test/java/api/requests/scenarios/RecallItemsTests.java (hunk 3 above)
Comment (Mykyta-Varenyk) [reply src/test/java/api/requests/scenarios/RecallItemsTests.java]: Fixed

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**